### PR TITLE
Add configuration option to keep secrets out of config

### PR DESCRIPTION
**Pull request by LenardHess**
Hi,

This pull request tackles issue #485.
I'm sending this in as an early first draft and to request comments.

EDIT: Updated revision is [in the comments](https://github.com/ddclient/ddclient/pull/522#issuecomment-1478677037), the rest of this initial comment is just kept for reference
---
My open points:
- Correct error handling when file is not found (instead of `open() or die`)
- Add tests for config parsing - I'm changing the password regex and want to be cautious.
- Either:
    - Mark password and password_file as needing either one or the other
    - Define placeholder value for password. If password_file is found and that placeholder value is not set, emit a warning.
- Make the password file less whitespace-sensitive (trailling newline)

This draft used the following config (domain references changed to example.com):
```
######################################################################
## Example configuration for password_file argument
######################################################################
daemon=300				# check every 300 seconds
syslog=yes				# log update msgs to syslog
#verbose=yes                            # Enable verbose logging
#mail=root				# mail all msgs to root
#mail-failure=root			# mail failed update msgs to root
pid=/var/run/ddclient/ddclient.pid	# record PID in file.
ssl=yes 				# use ssl-support.  Works with

# IP via fritzbox query script (local modified copy of sample-get-ip-from-fritzbox )
use=cmd, cmd=./get-ip-from-fritzbox-fixed

##
## NameCheap (namecheap.com)
##
protocol=namecheap,                       \
server=dynamicdns.park-your-domain.com,   \
login=example.com,                        \
password=iamnotused                       \
password_file=pw.secret                   \
@,subdomain.example.com
```
The file pw.secret contains the password without any extra whitespace (no trailing newlines)

### CODE REVIEW

**Comment by LenardHess** [comment]
An alternative approach that is more generic would be to allow any config key to be prefixed with 'file_'. The value would be the file path where to read the actual key's value.
(Inspired from [LinuxServer.io's approach to secrets](https://docs.linuxserver.io/images/docker-nextcloud#environment-variables-from-files-docker-secrets))

**Comment by SuperSandro2000** [comment]
I don't really have an opinion on this expect that using an env variable would probably the easiest solution.

I know that the config parsing logic is not that great nor has any test coverage but there are no plans to change that in the future due to the stopped development of the project.

**Comment by LenardHess** [comment]
Second version - this time with environment variables.

Instead of adding new config entries, this just adds a generic _env suffix for any configuration value.
This solves the open points of a missing file and the "password" keyword being required quirks of the first version.

Example configuration snippet:
```
##
## NameCheap (namecheap.com)
##
protocol=namecheap,                       \
server=dynamicdns.park-your-domain.com,   \
login_env=DD_LOGIN,                       \
password_env=DD_PASSWORD                  \
@
```
With this, ddclient will load the login and password from the environment variables DD_LOGIN and DD_PASSWORD respectively.

Remaining open points:
- Tests for config parsing
- Docs: Update comments, help output, add examples etc.

File: ddclient.in [hunk 1]
```diff
@@ -1508,7 +1508,7 @@ sub _read_config {
     # Configuration line format after comment and continuation
     # removal:
     #
-    #   [opt=value, ...] [host[, ...] [login [password]]]
+    #   [opt=value, ...] [host[, ...]]
```
Comment (SuperSandro2000) [inline ddclient.in:1511]: We should keep this, since it is technically still supported.

Comment (LenardHess) [reply ddclient.in]: Removed that commit

**Pull request closed without merging**